### DISCUSSION TITLE
Animate or transition only when element comes in viewport

### DISCUSSION THREAD

**Discussion by Devr-pro** [sveltejs/kit#8521] [Q&A]
Hey there!, I want to have a transition like scale,fade only when the element comes in viewport, like when user scrolls in, I may have solutions through libraries like GSAP,anime.js,etc but I want to implement it only through svelte code without any other package, How can I do so? Thanks.

**Comment by teemingc** [comment]
Check out these two libraries that implement the [`IntersectionObserver` API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) in Svelte. You can use them and they also act as good examples around this.
* https://svelte-intersection-observer.onrender.com
* https://www.npmjs.com/package/svelte-inview

**Reply by Devr-pro** [reply]
Alright, Thanks!
